pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -104,6 +108,23 @@ type SubnetDHCPConfig struct {
 	Mode DHCPConfigMode `json:"mode,omitempty"`
 }
 
+// AdvancedConfig is Subnet advanced configuration.
+type AdvancedConfig struct {
+	// StaticIPAllocation configuration for subnet ports with VIF attachment.
+	// If this field is not set, the static IP pool allocation will be automatically
+	// enabled when the DHCP mode is DHCPDeactivated or not set and disabled when the DHCP
+	// mode is DHCPServer or DHCPRelay.
+	// The static IP pool allocation and DHCP mode cannot both be enabled simultaneously
+	// in subnet.
+	StaticIPAllocation StaticIPAllocation `json:"staticIPAllocation,omitempty"`
+}
+
+// Static IP allocation for VPC Subnet ports with VIF attachement.
+type StaticIPAllocation struct {
+	// Activate or Deactivate static ip allocation for VPC Subnet ports with VIF attachement.
+	Enable *bool `json:"enable,omitempty"`
```

Comment:
The field is named "enabled" in NSX API, not "enable". Should it be consistent with NSX API?

Reply:
My bad. Thanks a lot for pointing it out!